### DISCUSSION TITLE
luminate\Contracts\Filesystem\Filesystem Error

### DISCUSSION THREAD

**Discussion by chibyk101** [spatie/laravel-backup#1710] [General]
I use PHP 8.1
Laravel Version 9.x

whenever I run the command "**php artisan backup:run**" I get the error

Spatie\Backup\BackupDestination\BackupDestination::__construct(): Argument #1 ($disk) must be of type ?Illuminate\Contracts\Filesystem\Filesystem, League\Flysystem\Filesystem given

**Comment by parallels999** [comment]
You need to indicate the package version
It looks like you upgrade from Laravel 8.x

**Reply by chibyk101** [reply]
Yes, it had something to do with the Laravel version. From Laravel 9.x there's a slight difference in extending the Storage facade.

This is connected to the spatie-dropbox-filesystem package rather than the spatie-laravel-backup package, so following the way it was extended from the docs fixed the issue. 

The example shown here [https://laravel.com/docs/10.x/filesystem#custom-filesystems](url) in the docs used spatie-dropbox-filesystem.
